src/main/java/org/kuali/kfs/kew/api/document/DocumentProcessingQueue.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * The Kuali Financial System, a comprehensive financial management system for higher education.
+ *
+ * Copyright 2005-2022 Kuali, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.kuali.kfs.kew.api.document;
+
+public interface DocumentProcessingQueue {
+
+    void process(String documentId) throws IllegalArgumentException;
+
+    void processWithOptions(String documentId, DocumentProcessingOptions options) throws IllegalArgumentException;
+
+    /*
+     * CU Customization: Added a separate method for handling the FINP-8541 functionality (which checks whether
+     * a document has been routed and, if not, will add a Complete request if one doesn't exist). This is needed
+     * to fix a DB transaction issue, where merely retrieving the route header can cause it to remain stale
+     * when subsequently invoking the workflow engine.
+     */
+    default boolean shouldProceedWithDocumentProcessing(String documentId) {
+        return true;
```

Comment:
maybe we should log this event as you mentioned it shouldn't happen and is a sign of something awry.  We can set up an alert in datadog if the log entry is found

Reply:
The problem with adding a log statement here is that this is an interface, not a class. As a result, a static `LOG` constant in this spot would have to be declared as public; Java doesn't let interfaces have private static constants.

If this default method somehow got invoked, I think the worst case is that the FINP-8541 feature won't be effective anymore. (That feature isn't a high priority; it just adds a Complete request to a SAVED document if such a request doesn't exist for some reason.) I doubt this default method would ever get invoked, and even if it did, it seems like the consequences of its execution would not be serious. Thus, we might not need to add logging here.

Thoughts?